cs/src/core/Index/FASTER/FASTER.cs

```diff
@@ -96,10 +98,30 @@ public FasterKV(long size, Functions functions, LogSettings logSettings, Checkpo
             CopyReadsToTail = logSettings.CopyReadsToTail;
             this.functions = functions;
 
+            if (logSettings.ReadCacheSizeBits > 0)
+            {
+                CopyReadsToTail = false;
+                UseReadCache = true;
+            }
+
             if (Utility.IsBlittable<Key>() && Utility.IsBlittable<Value>())
+            {
                 hlog = new BlittableAllocator<Key, Value>(logSettings, this.comparer);
+                if (UseReadCache)
+                {
+                    readcache = new BlittableAllocator<Key, Value>(logSettings, this.comparer);
+                    readcache.Initialize();
+                }
+            }
             else
+            {
                 hlog = new GenericAllocator<Key, Value>(logSettings, serializerSettings, this.comparer);
+                if (UseReadCache)
+                {
+                    readcache = new GenericAllocator<Key, Value>(logSettings, serializerSettings, this.comparer);
```

Comment:
>readcache = new GenericAllocator<Key, Value>(logSettings, serializerSettings, this.comparer); [](start = 20, length = 93)

I'm slightly worried about this impact this would have on the per-FASTER-instance footprint. Are there things we can slim down for the read cache allocator, or the original hlog allocator, in this use case?

Reply:
Yes, we should add the ability to provide a "settings" config for the read cache separately from the main hlog allocator (for page size, log size, etc.). This will allow us to size the read cache independently of hlog.

Reply:
Added settings config for read cache, and also allow users to control the read cache separately.